engine/gamesys/src/gamesys/scripts/script_physics.cpp

```diff
@@ -1398,6 +1398,14 @@ namespace dmGameSystem
         return 1;
     }
 
+    static void Physics_CheckIfShapeSizeValid(lua_State* L, float value)
+    {
+        if (value == 0.0f)
+        {
+            luaL_error(L,"Shape size can't be 0");
```

Comment:
Hm, this is just checking a float and doesn't say anything about the actual shape, I would probably make the function take a shape info as an argument, and rename it to CheckShape or something (the script library usually names functions Checkxxx where xxx is the type). Something like this perhaps? (pseudo-code)

```
void CheckShape(ShapeInfo info)
{
#define check_val(type_str, v) \
	if (fabs(v) < epsilon) \
		lua_error("%s ");

	switch(info.type)
	{
	case box:
		check_val("box.x", info.boxdimensions[0]);
		check_val("box.y", info.boxdimensions[1]);
		break;
	case sphere:
		check_val("sphere.radius", info.radius);
		break;
	etc..
	}
#undef check_val
}
```